rbi/core/enumerable.rbi

```diff
@@ -412,7 +412,7 @@ module Enumerable
 
   sig do
     params(
-        blk: T.proc.params(arg0: Elem).returns(BasicObject),
+        blk: T.proc.params(arg0: Elem).returns(T.any(Comparable, T::Array[Comparable])),
```

Comment:
This change could cause pay-server to fail, because it requires people to return a more narrow type from their block. Would you mind checking whether it passes? The steps for how to do it on a devbox are at http://go/types/local-dev

Reply:
Ah, actually since you made it from a fork one of the Sorbet team members will have to do it. @aisamanra can you takeover this PR?

Reply:
There are exactly 16 errors in pay-server, and all of them are from returning something that's `T.nilable(...)`. Presumably, it should be safe to `T.must` all of them: if they ever actually returned a `nil`, then `sort_by` would raise a runtime error! I'll toss together a pay-server PR that fixes them.

Reply:
Oh, some of the other errors _are_ going to cause problems. There's one instance, for example, of someone returning `T::Array[Integer, T::Array[String]]`, which doesn't work with these changes, either. @jez what would you think about getting this in with this method still expecting a `BasicObject` and then figuring out a more robust way of dealing with this in the future? Or, alternately, do you have good idea for a way of representing this type so that it can be more authentic to the Ruby semantics while still letting us get this PR in quickly?

Reply:
I guess it depends on how deep we want to go here (without changing the guts of how this works).

I tried this definition:

```
ComparableArray = T.type_alias(T::Array[T.any(Comparable, T::Array[Comparable])])
```

Which then allows the signatures to have:

```
T.any(Comparable, ComparableArray)
```

This is also not perfect because someone could return `T::Array[Integer, T::Array[T::Array[String]]` and that would still be valid Ruby, but not valid Sorbet 🤷‍♂

Reply:
Another (simpler) alternative is to just type it as `T.any(Enumerable, T::Array[BasicObject])` which would unblock the current cases and be slightly better than just going back to `BasicObject`

Reply:
I think that should work! It's probably a good idea to leave a comment on these just to note the way in which the type signatures are current stopgaps, as well, but that sounds like a good way forward.

Reply:
@aisamanra @jez ok, I ended up going the `T.any(Enumerable, T::Array[BasicObject])` route. What do you think?